proto/errors.go

```diff
@@ -73,16 +73,30 @@ var (
 type KafkaError struct {
 	errno   int16
 	message string
+	// An additional message explaining cases where the client returns this error internally
+	// instead of directly from a Kafka broker.
+	syntheticMessage string
 }
 
 func (err *KafkaError) Error() string {
+	if err.syntheticMessage != "" {
+		fmt.Sprintf("%s (%d %s)", err.message, err.errno, err.syntheticMessage)
+	}
 	return fmt.Sprintf("%s (%d)", err.message, err.errno)
 }
 
 func (err *KafkaError) Errno() int {
 	return int(err.errno)
 }
 
+func (err *KafkaError) SyntheticError() string {
+	return err.syntheticMessage
+}
+
+func (err *KafkaError) NewSynthetic(message string) *KafkaError {
+	return &KafkaError{err.errno, err.message, message}
```

Comment:
We talked f2f a bit but I don't remember if we got to a conclusion about the approach for the fix?